packages/app/src/hooks/useWelcomeSteps.tsx

```diff
@@ -139,16 +139,20 @@ export const stepsSelectors = {
 const ctx = createContext<Context>({} as Context);
 export function StepsProvider({ children }: WelcomeStepsProviderProps) {
   const navigate = useNavigate();
-  const [state, send, service] = useMachine<Machine>(
-    welcomeStepsMachine.withConfig({
-      actions: {
-        navigateTo: (context) => {
-          if (context.current.id > 2) return;
-          navigate?.(`/welcome/${context.current.path}`);
+
+  const machine = useMemo(
+    () =>
+      welcomeStepsMachine.withConfig({
```

Comment:
dude, you don't need `useMemo` inside this, it's already has all performance handles inside `useMachine`

Reply:
we needed because without useMemo we had errors in console saying that machine was different between renders. 

but I removed now to check and the errors are not happening anymore. removing it.